src/server/modules/dao/ModDAO.ts

```diff
@@ -26,12 +33,29 @@ export default class ModDAO extends BaseDAO<IDbMod> implements IDbModDAO {
                 .split("@")[0]
                 .trim()
         }));
-        const foundDependencies = await await this.collection.find({ $or: _dependencies }).toArray();
-        if (foundDependencies.length !== _dependencies.length) {
+        const foundDependencies = await this.collection.find({ $or: _dependencies }).toArray();
+        const deDupedFound = await this.dedupeDependencies(foundDependencies);
+        if (deDupedFound.length !== _dependencies.length) {
             throw new ServerError("server.invalid_dependencies", [], 400);
         }
+
         return foundDependencies;
```

Comment:
Shouldn't this return `deDupedFound`? I think the return value only matters for the API endpoint though.

Reply:
Actually now that I think about it, returning deDupedFound might cause inconsistencies depending on which API endpoint you use to get the dependencies. When you just get the mod, the dependency list is generated by a MongoDB query elsewhere that doesn't have deduping. IDK what mod installers are doing though.

Reply:
I know ModAsistant is doing dependency resolution independently because beatmods doesn't do it well